### DISCUSSION TITLE
How to update many package version and push to git concurrentl

### DISCUSSION THREAD

**Discussion by ngoctuan001** [semantic-release/semantic-release#1789] [Q&A] [answered]
## New feature motivation

So i'm doing a development using lerna monorepo. In our repo, lets say we have 2 packages:

--packages
-----chart
-----ui

Then in each package, this is my `release.config.js`

**chart**
```
module.exports = {
  plugins: [
    [
      '@semantic-release/commit-analyzer',
      {
        preset: 'angular',
        parserOpts: {
          headerPattern: /^(\w*)(\((chart|\*)\))?: (.*)$/,
        },
      },
    ],
    [
      '@semantic-release/release-notes-generator',
      {
        preset: 'angular',
        parserOpts: {
          headerPattern: /^(\w*)(\((chart|\*)\))?: (.*)$/,
          headerCorrespondence: ['type', 'scope'],
        },
      },
    ],
    '@semantic-release/changelog',
    '@semantic-release/npm',
    [
      '@semantic-release/git',
      {
        assets: ['package.json'],
        message: 'chore(release): <%= nextRelease.gitTag %> - [skip ci]\\n\\n<%= nextRelease.notes %>',
      },
    ],
  ],
};

```

**ui**
```
module.exports = {
  plugins: [
    [
      '@semantic-release/commit-analyzer',
      {
        preset: 'angular',
        parserOpts: {
          headerPattern: /^(\w*)(\((ui|\*)\))?: (.*)$/,
        },
      },
    ],
    [
      '@semantic-release/release-notes-generator',
      {
        preset: 'angular',
        parserOpts: {
          headerPattern: /^(\w*)(\((ui|\*)\))?: (.*)$/,
          headerCorrespondence: ['type', 'scope'],
        },
      },
    ],
    '@semantic-release/changelog',
    '@semantic-release/npm',
    [
      '@semantic-release/git',
      {
        assets: ['package.json'],
        message: 'chore(release): <%= nextRelease.gitTag %> - [skip ci]\\n\\n<%= nextRelease.notes %>',
      },
    ],
  ],
};

```

So as you can see, here i use commit analyzer to pick out the correct scope for the current releasing package, then release it to npm and **push new version in package.json to git**.

So there is no problem if the pipeline only release 1 package. If 2 packages release at the same time, the later package will fails, because it **cannot push new version to git**. The reason is because the former one already pushed and update the git head to new head, making the later one fails.

Is there anyway to implement concurrent publish/ git push. Thanks

**Comment by travi** [comment, marked as answer]
semantic-release provides no official support for packages managed within monorepos. the types of complexities that you describe are along the lines of the reasons why we have not added support for this structure.

there are some tools that add support for specific approaches for managing packages withing a monorepo on top of semantic-release, but support for those additional tools is provided by their maintainers, and not the core team here. one that i have used with success in a lerna mnaaged monorepo is [semantic-release-monorepo](https://github.com/pmowrer/semantic-release-monorepo). however, even that one recommends running against the multiple packages serially.
